### DISCUSSION TITLE
Mechanism automatically make breaking changes report for vendor

### DISCUSSION THREAD

**Discussion by areknoster** [areknoster/hypert#8] [Ideas]
Hypert statically records API interactions. The issue we often face is that vendors sometimes unintentionally introduce breaking changes.

I'm wondering if it would be worthy to implement a way to automatically generate breaking changes report to be sent to the vendor. I imagine the workflow to be among the lines of:
1. I recorded working interactions with some vendor's API
2. After some time my code starts giving out unexpected errors. Hypert tests in replay mode are still passing
3. I rerun test in 'Report' mode. The report contains old recorded requests and responses and the new ones that made my code break. Optionally we could think of some mechanism to detect what was the breaking change

What do you think? CC @pnmcosta

**Comment by pnmcosta** [comment]
Umm, one would hope that vendors are using standards and breaking changes aren't introduced to existing endpoints.

Based on my experience, consuming mostly REST endpoints, new endpoints (e.g. v2/v3/whatnot) are introduced to prevent this.

Most issues with existing endpoints IMO are transient and/or auth related, would a Report be truly useful?

Then again, I may be missing something :D

**Comment by areknoster** [comment]
Yes, that would be ideal if breaking changes were always introduced with new major version, but mistakes happen and not all vendors govern their APIs so well. 
I think I might be skewed by one vendor I happen to work with
